constellation_base/models.py

```diff
@@ -39,3 +42,19 @@ def settings_dict(self):
             'organization': self.organization,
             'titlebar_links': self.titlebar_links,
         }
+
+
+class LocalGroupACLEntry(models.Model):
+    """ACL for Local Groups that don't exist in other authentication sources"""
+    group = models.ForeignKey(Group)
+
+    # This is a string since the user might not exist at the time they log in
+    # and need this to be applied
+    username = models.CharField(max_length=150)
```

Comment:
models.py
```diff
-group = models.ForeignKey(Group)
+groups = models.ManyToManyField(Group)
```

views.py
```diff
for member in members:
-    lgacle = LocalGroupACLEntry()
+    lgacle = LocalGroupACLEntry.objects.get_or_create(username=member)
-    lgacle.username = member
-    lgacle.group = group
+    lgacl.groups.add(group)
     lgacle.save()
```

Finally, in signals you can do:
```diff
-supGroups = LocalGroupACLEntry.objects.filter(username=user.username)
+supGroups = LocalGroupACLEntry.objects.get(username=user.username).groups.all()
```
You probably ought to add a uniqueness constraint on username at this point...